vkubelet/pod.go

```diff
@@ -218,11 +303,64 @@ func (s *Server) updatePodStatus(ctx context.Context, pod *corev1.Pod) error {
 	return nil
 }
 
-func (s *Server) enqueuePodStatusUpdate(ctx context.Context, q workqueue.RateLimitingInterface, pod *corev1.Pod) {
+func (s *Server) updateRawPodStatus(ctx context.Context, pod *corev1.Pod) error {
+	ctx, span := trace.StartSpan(ctx, "updateRawPodStatus")
+
+	// Since our patch only applies to the status subtype, we should be safe in doing this
+	// We don't really have a better option, as this method is only called from the async pod notifier,
+	// which provides (ordered)
+	pod.ObjectMeta.ResourceVersion = ""
+	patch, err := json.Marshal(pod)
+	if err != nil {
+		span.SetStatus(ocstatus.FromError(err))
+		return pkgerrors.Wrap(err, "Unable to serialize patch JSON")
+	}
+
+	newPod, err := s.k8sClient.CoreV1().Pods(pod.Namespace).Patch(pod.Name, types.MergePatchType, patch, "status")
```

Comment:
This should restrict the patch from only applying to "status" subresource. I'm not 100% sure about this behaviour as the documentation on merge patch (non-strategic) + subresources seems to be...vague.